src/sels_fe/src/components/quiz/Quiz/QuizWithQuestionsCreate.jsx

```diff
@@ -0,0 +1,305 @@
+  postQuiz,
+  updateQuestion,
+  updateQuiz,
+} from "../../../store/actions"
+
+import { Stepper, StepButton, Step, Box } from "@mui/material"
+import { IconButton, Button, Typography } from "@material-ui/core"
+
+import { Delete } from "@material-ui/icons"
+import { Divider, Stack } from "@mui/material"
+
+import QuizForm from "./components/QuizForm"
+import QuestionsForm from "./../Question/components/QuestionForm"
+import ChoiceForm from "./../Choice/components/ChoiceForm"
+import AnswersSelect from "../Answer/components/AnswersSelect"
+
+const steps = [
+  "Add Quiz Details",
+  "Add Questions",
+  "Add Choices",
+  "Select Answers",
+]
+let currentQuiz = null
+let currentQuestions = null
+
+function QuizWithQuestionsCreate() {
+  const quizzes = useSelector((state) => state.Quiz.quiz_list)
+  const question_list = useSelector((state) => state.Question.question_list)
+  const choices = useSelector((state) => state.Choice.choice_list)
+  let errorQuiz = useSelector((state) => state.Quiz.error)
+
+  const [activeStep, setActiveStep] = useState(0)
+  const [completed, setCompleted] = useState([])
+  const [update, setUpdate] = useState(false)
+
+  const [quiz, setQuiz] = useState({ name: "", description: "" })
+  const [questions, setQuestions] = useState([
+    { question: "", description: "", quiz: [] },
+  ])
+  const [questionAnswer, setQuestionAnswer] = useState([
+    { choice: 0, question: 0 },
+  ])
+
+  const dispatch = useDispatch()
+  const history = useHistory()
+
+  const handleQuizChange = (event) => {
+    const { name, value } = event.target
+    setQuiz({ ...quiz, [name]: value })
+  }
+
+  const handleQuestionsChange = (index, event) => {
+    let questionsFormVal = [...questions]
+    questionsFormVal[index][event.target.name] = event.target.value
+    setQuestions(questionsFormVal)
+  }
+
+  let addQuestionsForm = () => {
+    setQuestions([...questions, { question: "", description: "", quiz: [] }])
+  }
+
+  let removeQuestionsForm = (index) => {
+    let questionsFormValue = [...questions]
+    questionsFormValue.splice(index, 1)
+    setQuestions(questionsFormValue)
+  }
+
+  let handleSetQuestionAnswer = (event) => {
+    const { name, value } = event.target
+    setQuestionAnswer({ ...questionAnswer, [name]: value })
+  }
+
+  const handleSubmit = () => {
+    if (activeStep === 0) {
+      completed[0] !== true
+        ? dispatch(postQuiz(quiz))
+        : dispatch(updateQuiz(currentQuiz[0]?.id, quiz))
+    }
+    if (activeStep === 1) {
+      questions.map((question, index) => {
+        const question_id = new Array()
+        question_id.push(currentQuiz[0]?.id)
+        question["quiz"] = question_id
+        completed[1] !== true
+          ? dispatch(postQuestion(question))
+          : dispatch(updateQuestion(currentQuestions[index]?.id, question))
+      })
+    }
+  }
+
+  useEffect(() => {
+    currentQuiz = quizzes.filter((q) => q.name === quiz.name)
+    if (currentQuiz[0]?.id) handleComplete()
```

Comment:
Confirmation: Why is the first item of `currentQuiz` being checked?
When I tried to check behavior `handleComplete()` was being called twice, since its triggered with Button click as well.

Reply:
Reason for check is the one below, It's not using state so the value
was mutated with every rerender. Done with the update.